File: find-identities.py
#!/usr/bin/env python3
import sys
import json
import yaml

# always only one line from terraform
# {"env":"sandbox"}
line = sys.stdin.readline()
query = json.loads(line)


def filter_environment():
    # for identity in identity_mapping:
    #     for environment in identity:
    #         if query['management_group'] in environment.management_group:
    #             add to result
    return filter(lambda x: (query['env'] for x['name']['environments'] in x['environments']), identity_config['mappings'])


with open("identity.yaml", 'r') as stream:
    identity_config = yaml.load(stream, Loader=yaml.FullLoader)

    result = {
        'identities': ';'.join([x['name'] for x in filter_environment()]),
        'keyvault_names': ';'.join(["{}-{}".format(x['keyvault']['name'], query['management_group'].lower()) for x in filter_environment()]),
        'keyvault_rgs': ';'.join(["{}-{}".format(x['keyvault']['resource_group'], query['management_group'].lower()) for x in filter_environment()]),
        'keyvault_subscription_ids': ';'.join(["{}-{}".format(x['keyvault']['resource_group'], query['management_group'].lower()) for x in filter_environment()]),
    }

    print(json.dumps(result))

File: find-identities.js
#!/usr/bin/env node

const fs = require('fs')
const YAML = require('yaml')

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin
})

// always only one line from terraform
// {"env":"sandbox"}
rl.on('line', (line) => {
    const env = JSON.parse(line).env

    const file = fs.readFileSync('./identity.yaml', 'utf8')

    const message = YAML.parse(file);

    const filtered_map = message.mappings
        .map(mapping => {
                return mapping.environments
                    .map(environment => {
                            return {
                                env: environment.name,
                                subscription_id: environment.subscription_id,
                                name: `${mapping.name}-${environment.name}`,
                                keyvault_name: `${mapping.keyvault.name}-${environment.name}`,
                                keyvault_rg: `${mapping.keyvault.resource_group}-${environment.name}`
                            }
                        }
                    )

            }
        )
        .flat()
        .filter(mapping => mapping.env === env)

    const result = {
        subscription_ids: filtered_map.map(mapping => mapping.subscription_id).join(';'),
        identities: filtered_map.map(mapping => mapping.name).join(';'),
        keyvault_names: filtered_map.map(mapping => mapping.keyvault_name).join(';'),
        keyvault_rgs: filtered_map.map(mapping => mapping.keyvault_rg).join(';'),
    }

    console.log(JSON.stringify(result))
})